cpp/tests/io/arrow_io_source_test.cpp

```diff
@@ -87,19 +88,16 @@ TEST_F(ArrowIOTest, S3FileSystem)
     ASSERT_EQ(1, tbl.tbl->num_columns());  // Only single column specified in reader_options
     ASSERT_EQ(244, tbl.tbl->num_rows());   // known number of rows from the S3 file
   }
+
+#ifdef ARROW_S3
   if (!s3_unsupported) {
     // Verify that we are using Arrow with S3, and call finalize
     // https://github.com/apache/arrow/issues/36974
     // This needs to be in a separate conditional to ensure we call
     // finalize after all arrow_io_source instances have been deleted.
-    void* whole_app                                       = dlopen(NULL, RTLD_LAZY);
-    decltype(arrow::fs::EnsureS3Finalized)* close_s3_func = nullptr;
-
-    close_s3_func = reinterpret_cast<decltype(close_s3_func)>(
-      dlsym(whole_app, "_ZN5arrow2fs17EnsureS3FinalizedEv"));
-    if (close_s3_func) { CUDF_EXPECTS(close_s3_func().ok(), "Failed to finalize s3 filesystem"); }
-    dlclose(whole_app);
+    arrow::fs::EnsureS3Finalized();
```

Comment:
Probably need this to fix the build warning from `[[nodiscard]]`. Or you can check the status explicitly (not sure exactly how to write that without more thinking).
```suggestion
    [[maybe_unused]] arrow::fs::EnsureS3Finalized();
```